alpine/packagescanner.go

```diff
@@ -119,7 +119,12 @@ func (*Scanner) Scan(ctx context.Context, layer *claircore.Layer) ([]*claircore.
 		pkgs = append(pkgs, &p)
 	}
 	if err := s.Err(); err != nil {
-		return nil, err
+		if len(pkgs) == 0 {
```

Comment:
what's the idea here? if you've already scanned some packages, ignore the scan error?
we've been following a "if any error, fail hard" policy - should we not here?